crates/cxx-qt-lib/src/core/qanystringview.rs

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Joshua Goins <josh@redstrate.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+use cxx::{type_id, ExternType};
+use std::ffi::c_void;
+use std::mem::MaybeUninit;
+
+#[cxx::bridge]
+mod ffi {
+    #[namespace = "Qt"]
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qt.h");
+    }
+
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qanystringview.h");
+        type QAnyStringView = super::QAnyStringView;
+
+        /// Returns true if the string has no characters; otherwise returns false.
+        #[rust_name = "is_empty"]
+        fn isEmpty(self: &QAnyStringView) -> bool;
+
+        /// Returns true if this string is null; otherwise returns false.
+        #[rust_name = "is_null"]
+        fn isNull(self: &QAnyStringView) -> bool;
+    }
+
+    #[namespace = "rust::cxxqtlib1"]
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/common.h");
+
+        #[doc(hidden)]
+        #[rust_name = "QAnyStringView_drop"]
+        fn drop(string: &mut QAnyStringView);
+
+        #[doc(hidden)]
+        #[rust_name = "QAnyStringView_init_default"]
+        fn construct() -> QAnyStringView;
+        #[doc(hidden)]
+        #[rust_name = "QAnyStringView_init_from_rust_string"]
+        fn qanystringviewInitFromRustString(string: &str) -> QAnyStringView;
+        #[doc(hidden)]
+        #[rust_name = "QAnyStringView_init_from_QAnyStringView"]
+        fn construct(string: &QAnyStringView) -> QAnyStringView;
+
+        #[doc(hidden)]
+        #[rust_name = "QAnyStringView_eq"]
+        fn operatorEq(a: &QAnyStringView, b: &QAnyStringView) -> bool;
+        #[doc(hidden)]
+
+        #[doc(hidden)]
+        #[rust_name = "QAnyStringView_len"]
+        fn qanystringviewLen(string: &QAnyStringView) -> isize;
+    }
+}
+
+/// The QAnyStringView class provides a Unicode character string.
+///
+/// Note that QAnyStringView is a UTF-16 whereas Rust strings are a UTF-8
+#[repr(C)]
+pub struct QAnyStringView {
```

Comment:
add an entry to the changelog to mention that you have added a new type :-)

Reply:
Done